keanu-project/src/main/java/io/improbable/keanu/network/LambdaSection.java

```diff
@@ -117,6 +160,7 @@ public static LambdaSection getDownstreamLambdaSection(Vertex<?> aVertex, boolea
                     queued.add(next);
                 }
             }
+            visited.add(visiting);
```

Comment:
If you don't add the visiting until the end here you miss adding observed and probabilistic to the visited set.